[docs]: +zi-message usage + zsh-zoxide (#59)
z-shell/wiki: ecosystem/plugins/zsh_zoxide.md

---
id: zsh-zoxide
title: ⚙️ Zoxide

> image: img/logo/320x320.png

Reviewer: ```suggestion
image: /img/logo/320x320.png
```

description: The ajeetdsouza/zoxide init for Zsh
keywords:
  - zsh-plugin
  - zsh-zoxide
  - zoxide
  - cd
---

<!-- @format -->

- [z-shell/zsh-zoxide](https://github.com/z-shell/zsh-zoxide)

[`ajeetdsouza/zoxide`](https://github.com/ajeetdsouza/zoxide) is a smarter `cd` command, inspired by `z` and `autojump`.

> It remembers which directories you use most frequently, so you can "jump" to them in just a few keystrokes.

Reviewer: ```suggestion
It remembers which directories you use most frequently, so you can "jump" to them in just a few keystrokes.
```

zoxide works on all major shells.

> ### Install `zoxide`

Reviewer: ```suggestion
## Install `zoxide`
```

- [Official install](https://github.com/ajeetdsouza/zoxide#step-1-install-zoxide) (recommended)
- With Zi:

> ```zsh

Reviewer: ```suggestion
```shell showLineNumbers
```

zi ice as'null' from"gh-r" sbin
zi light ajeetdsouza/zoxide
```

> ### Install `zsh-zoxide`

Reviewer: ```suggestion
## Install `zsh-zoxide`
```

> #### Standart syntax

Reviewer: ```suggestion
### Standart syntax
```

> ```zsh

Reviewer: ```suggestion
```shell showLineNumbers
```

zi ice has'zoxide'
zi light z-shell/zsh-zoxide
```

> #### The "For" syntax

Reviewer: ```suggestion
### The "For" syntax
```

> ```zsh

Reviewer: ```suggestion
```shell showLineNumbers
```

zi has'zoxide' light-mode for \
  z-shell/zsh-zoxide
```

> #### Turbo mode + "For" syntax

Reviewer: ```suggestion
### Turbo mode + "For" syntax
```

> ```zsh

Reviewer: ```suggestion
```shell showLineNumbers
```

zi has'zoxide' wait lucid for \
  z-shell/zsh-zoxide
```

> > Wiki: [automatic, condition based (loading/unloading)](https://wiki.zshell.dev/docs/getting_started/overview#automatic-condition-based---load--unload)

Reviewer: ```suggestion
> Wiki: [automatic, condition based (loading/unloading)](/docs/getting_started/overview#automatic-condition-based---load--unload)
```

### Add interactive selection with [`fzf`](https://github.com/z-shell/fzf)

> [fzf](https://github.com/junegunn/fzf) is a command-line fuzzy finder, used by zoxide for interactive selection. It can be installed from [here](https://github.com/z-shell/fzf) as a zi package.

Reviewer: ```suggestion
[fzf](https://github.com/junegunn/fzf) is a command-line fuzzy finder, used by zoxide for interactive selection. It can be installed as a Zi [package](/ecosystem/packages/available-packages#the-fzf-command-line-fuzzy-finder).
```

> ### Environment variables and usage with Zi

Reviewer: ```suggestion
## Environment variables and usage with Zi
```

The plugin will call `zoxide init` with prefixed commands `x`, `xi`:

> - [x] Completions auto-loaded: [commands](https://wiki.zshell.dev/docs/guides/commands#completions-management), [ice-modifiers](https://wiki.zshell.dev/docs/guides/syntax/ice-modifiers#completions)

Reviewer: ```suggestion
- [x] Completions auto-loaded: [commands](/docs/guides/commands#completions-management), [ice-modifiers](/docs/guides/syntax/ice-modifiers#completions)
```

> - [x] Manpages auto installed: [\$ZI[MAN_DIR]](https://wiki.zshell.dev/docs/guides/customization#customizing-paths)

Reviewer: ```suggestion
- [x] Manpages auto installed: [\$ZI[MAN_DIR]](/docs/guides/customization#customizing-paths)
```

- [x] [Database](https://github.com/ajeetdsouza/zoxide#environment-variables) directory set: [\$ZPFX/share](https://wiki.zshell.dev/community/zsh_plugin_standard#global-parameter-with-prefix), [customizing-paths](https://wiki.zshell.dev/docs/guides/customization#customizing-paths)

> ```sh

Reviewer: ```suggestion
```sh showLineNumbers
```

x foo              # cd into highest ranked directory matching foo
x foo bar          # cd into highest ranked directory matching foo and bar
x foo /            # cd into a subdirectory starting with foo
```

> ```sh

Reviewer: ```suggestion
```sh showLineNumbers
```

x ~/foo            # z also works like a regular cd command
x foo/             # cd into relative path
x ..               # cd one level up
x -                # cd into previous directory
```

```sh
xi foo             # cd with interactive selection (using fzf)
```

```sh
x foo<SPACE><TAB>  # show interactive completions
```

> #### Change prefix

Reviewer: ```suggestion
### Change prefix
```

If you want to replace `cd` with `zoxide`. You can set the prefix to `cd` by setting `_ZO_CMD_PREFIX=cd` in `~.zshrc`

> before the installation of zoxide

Reviewer: ```suggestion
If you want to replace `cd` with `zoxide`. You can set the prefix to `cd` by setting `_ZO_CMD_PREFIX=cd` in `~.zshrc` before the installation of zoxide
```

> ### Environment variables and usage with other plugin managers

Reviewer: ```suggestion
## Environment variables and usage with other plugin managers
```

The plugin will call `zoxide init` with prefixed commands `z`, `zi`.

> ### Import your data

Reviewer: ```suggestion
## Import your data
```

If you currently use any of the following utilities, you may want to import your data into zoxide:

> #### autojump

Reviewer: ```suggestion
### autojump
```

```shell
x import --from autojump path/to/db # replace x with your alias
```

> #### z, z.lua or zsh-z

Reviewer: ```suggestion
### z, z.lua or zsh-z
```

---
id: zsh-zoxide
title: ⚙️ Zoxide
image: /img/logo/320x320.png
description: The ajeetdsouza/zoxide init for Zsh
keywords:
  - zsh-plugin
  - zsh-zoxide
  - zoxide
  - cd
---

<!-- @format -->

import Tabs from '@theme/Tabs';
import TabItem from '@theme/TabItem';

- [z-shell/zsh-zoxide](https://github.com/z-shell/zsh-zoxide)

[`ajeetdsouza/zoxide`](https://github.com/ajeetdsouza/zoxide) is a smarter `cd` command, inspired by `z` and `autojump`.

`zoxide` is a smarter `cd` command written in Rust which remember most used directory and add the ability to "jump" inside without typing the entire path.
`zoxide` also provide interactive completion with `fzf`

[![asciicast](https://asciinema.org/a/512856.svg)](https://asciinema.org/a/512856)

## Install `zoxide`

First, you need to have `zoxide` installed on your system. You can install using two differents ways:
<Tabs>
<TabItem value="official" label="Official install" default>

Follow [https://github.com/ajeetdsouza/zoxide#step-1-install-zoxide](https://github.com/ajeetdsouza/zoxide#step-1-install-zoxide)

</TabItem>

> <TabItem value="zi" label="WIth Zi" default>

Reviewer: ```suggestion
<TabItem value="zi" label="With Zi" default>
```
